### PR TITLE
Button: rename buttons & deprecate props 

### LINKED ISSUE #308: Button: v5 breaking changes
- [ ] Rename the button (`variant` prop)
- [ ] Remove the deprecated `theme`
- [ ] Remove the drprecated`Secondary Button`

### PR DESCRIPTION
**Pull request by rosslo**
# What changed
- Rename the values of `variant` prop and combine it with removed `theme` prop
   - `varaint`: "default" & `theme`: "white"-> `variant`: "solid-white"
   - `varaint`: "default" & `theme`: "blue"-> `variant`: "solid-blue"
   - `varaint`: "primary" -> `variant`: "yellow"
- Rename the `className` of `SolidButton` and `SolidShadowButton`
   - `SolidButton`: "aries-defaultbtn" -> "aries-solid-btn"
   - `SolidShadowButton`: "aries-primarybtn" -> "aries-solid-shadow-btn";
- Add new button `variant`: "white-grey"
- Remove deprecated `theme`
   - solid button: "red", "yello"
   - solid-shadow button: 'red', 'blue', 'blue-red'
- Remove deprecated SecondaryButton
- Remove useless `removeHoverEffect` prop

### CODE REVIEW

File: stories/General/ButtonStory.tsx [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 import {
   ViewIcon,
   ArrowRoundForwardIcon,
 } from '../../src/General/Icon/components';
 
-import { Variant, Theme } from '../../src/Utils/StyleConfig';
+import { ButtonVariant } from '../../src/Utils/StyleConfig';
+
+const ButtonVariantWithoutLink = omit(ButtonVariant, ['LINK']);
+const AllButtonNamesWithouLink =
```
Comment (westwood846) [inline stories/General/ButtonStory.tsx]: Typo here

**Review by westwood846 — CHANGES_REQUESTED** [review]
Mostly LGTM, there's just a small typo in the ButtonStory.

About _solid-white_ and _white-grey_ buttons: The only difference between them is whether they have the hover effect or not. Could we keep `removeHoverEffect` and get rid of _white-grey_? Also, I've seen something like _white-grey_ used on grey backgrounds, maybe this should rather be a transparent background button?

**Comment by rosslo** [comment]
> About _solid-white_ and _white-grey_ buttons: The only difference between them is whether they have the hover effect or not. Could we keep `removeHoverEffect` and get rid of _white-grey_? Also, I've seen something like _white-grey_ used on grey backgrounds, maybe this should rather be a transparent background button?

I don't think `removeHoverEffect` to control style will be a good way, I'd prefer to creating a new component then simply use `variant` to control style instead. 

For example, I also removed the `theme` prop to use `variant` only to control the button UI.

One benefit from it is we can use the same name as GDS's to help the communication between the devs and designers.

https://app.abstract.com/projects/da6f9bdf-2057-45bf-9dda-8dbf0d261eac/branches/master/commits/c28ce4aa716f49c2dd8a7671e4f497c159b43301/files/DDF7C217-39F9-45D7-8C53-8D8C88D5FC37/layers/083D78C7-FD27-47F5-B5BA-19041A61E7C1?mode=design
image

**Comment by westwood846** [comment]
> I don't think `removeHoverEffect` to control style will be a good way, I'd prefer to creating a new component then simply use `variant` to control style instead.
> 
> For example, I also removed the `theme` prop to use `variant` only to control the button UI.
> 
> One benefit from it is we can use the same name as GDS's to help the communication between the devs and designers.

I see. I agree that we should stay consistent with GDS and I love that the button situation is getting simplified. I just foresee that there will be more variants added in the future that are jsut added to get rid of the hover.

**Review by westwood846 — APPROVED** [review]
LGTM

**Pull request merged**